Framework/PythonInterface/plugins/algorithms/HB3AIntegratePeaks.py

```diff
@@ -0,0 +1,113 @@
+                             doc="Peaks workspace containing peaks to integrate")
+
+        positive_val = FloatBoundedValidator(lower=0.0)
+        self.declareProperty("PeakRadius", defaultValue=1.0, validator=positive_val,
+                             doc="Fixed radius around each peak position in which to integrate"
+                                 " (same units as input workspace) ")
+
+        self.declareProperty("BackgroundInnerRadius", defaultValue=0.0, validator=positive_val,
+                             doc="Inner radius used to evaluate the peak background")
+        self.declareProperty("BackgroundOuterRadius", defaultValue=0.0, validator=positive_val,
+                             doc="Outer radius used to evaluate the peak background")
+
+        formats = StringListValidator()
+        formats.addAllowedValue("SHELX")
+        formats.addAllowedValue("Fullprof")
+        self.declareProperty("OutputFormat", defaultValue="SHELX", validator=formats,
+                             doc="Save direction cosines in HKL, or the fullprof format")
+
+        self.declareProperty(FileProperty(name="OutputFile", defaultValue="",
+                                          direction=Direction.Input,
+                                          action=FileAction.Save),
+                             doc="Filepath to save the integrated peaks workspace in HKL format")
+
+        self.declareProperty(IPeaksWorkspaceProperty("OutputWorkspace", defaultValue="", direction=Direction.Output,
+                                                     optional=PropertyMode.Mandatory),
+                             doc="Output peaks workspace (copy of input with updated peak intensities)")
+
+    def validateInputs(self):
+        issues = dict()
+
+        # Make sure outer radius > inner radius
+        inner_radius = self.getProperty("BackgroundInnerRadius").value
+        outer_radius = self.getProperty("BackgroundOuterRadius").value
+
+        if outer_radius < inner_radius:
+            issues['BackgroundOuterRadius'] = "Outer radius should be >= to inner radius"
+
+        return issues
+
+    def PyExec(self):
+        input_ws = self.getProperty("InputWorkspace").value
+        peak_ws = self.getProperty("PeaksWorkspace").value
+
+        peak_radius = self.getProperty("PeakRadius").value
+        inner_radius = self.getProperty("BackgroundInnerRadius").value
+        outer_radius = self.getProperty("BackgroundOuterRadius").value
+
+        out_format = self.getProperty("OutputFormat").value
+        filename = self.getProperty("OutputFile").value
+
+        out_ws = IntegratePeaksMD(InputWorkspace=input_ws,
+                                  PeakRadius=peak_radius,
+                                  BackgroundInnerRadius=inner_radius,
+                                  BackgroundOuterRadius=outer_radius,
+                                  PeaksWorkspace=peak_ws)
+
+        # Apply Lorentz correction:
+        for p in range(out_ws.getNumberPeaks()):
```

Comment:
I think we should make the Lorentz correction optional, the default can be true.